src/query_plan/display.rs

```diff
@@ -7,29 +7,40 @@ pub(crate) struct State<'fmt, 'fmt2> {
     output: &'fmt mut fmt::Formatter<'fmt2>,
 }
 
-impl State<'_, '_> {
-    fn write<T: fmt::Display>(&mut self, value: T) -> fmt::Result {
+impl<'a, 'b> State<'a, 'b> {
+    pub(crate) fn new(output: &'a mut fmt::Formatter<'b>) -> State<'a, 'b> {
```

Comment:
This is more for a hygiene follow up in the future. It seems like we now how specific display impls in `query_plan` and `query_graph` modules. I like that `query_plan` groups it all together in `display.rs`, `query_graph` doesn't have that. I think we can have this `State` live outside of the two modules (straight up in `src`?) and have `display.rs` be implemented in both.